src/vm/class.cpp

```diff
@@ -3767,6 +3767,10 @@ VOID EEClassLayoutInfo::CollectLayoutFieldMetadataThrowing(
     {
         classSizeInMetadata = 0;
     }
+    else
```

Comment:
Please comment on the logic here. If we didn't fail to get the size, this indicates we have an explicit layout? Which would seem to imply we can check `cl` much earlier.

Reply:
It's not explicit layout, it's explicitly defined size (like `[StructLayout(..., Size = N)]`. I'll add a comment.